src/integrationTest/sam.test.ts

```diff
@@ -5,46 +5,78 @@
-    // { name: 'python2.7', path: 'testProject/hello_world/app.py', debuggerType: 'python' },
-    // { name: 'python3.6', path: 'testProject/hello_world/app.py', debuggerType: 'python' },
-    // { name: 'python3.7', path: 'testProject/hello_world/app.py', debuggerType: 'python' },
-    // { name: 'python3.8', path: 'testProject/hello_world/app.py', debuggerType: 'python' }
-    // { name: 'dotnetcore2.1', path: 'testProject/src/HelloWorld/Function.cs', debuggerType: 'coreclr' }
+
+interface TestScenario {
+    runtime: Runtime
+    path: string
+    debugSessionType: string
+    language: Language
+}
+
+interface LocalInvokeCodeLensCommandResult {
+    datum: Datum
+}
+
+const scenarios: TestScenario[] = [
+    {
+        runtime: 'nodejs8.10',
+        path: 'hello-world/app.js',
+        debugSessionType: 'node2',
+        language: 'javascript'
+    },
+    {
+        runtime: 'nodejs10.x',
+        path: 'hello-world/app.js',
+        debugSessionType: 'node2',
+        language: 'javascript'
+    },
+    { runtime: 'nodejs12.x', path: 'hello-world/app.js', debugSessionType: 'node2', language: 'javascript' }
+    // { runtime: 'python2.7', path: 'hello_world/app.py', debugSessionType: 'python' },
+    // { runtime: 'python3.6', path: 'hello_world/app.py', debugSessionType: 'python' },
+    // { runtime: 'python3.7', path: 'hello_world/app.py', debugSessionType: 'python' },
+    // { runtime: 'python3.8', path: 'hello_world/app.py', debugSessionType: 'python' }
+    // { runtime: 'dotnetcore2.1', path: 'src/HelloWorld/Function.cs', debugSessionType: 'coreclr' }
 ]
 
-async function openSamProject(projectPath: string): Promise<vscode.Uri> {
-    const documentPath = path.join(projectFolder, projectPath)
-    const document = await vscode.workspace.openTextDocument(documentPath)
+async function openSamAppFile(applicationPath: string): Promise<vscode.Uri> {
+    const document = await vscode.workspace.openTextDocument(applicationPath)
 
     return document.uri
 }
 
-function tryRemoveProjectFolder() {
+function tryRemoveFolder(fullPath: string) {
     try {
-        removeSync(path.join(projectFolder, 'testProject'))
-    } catch (e) {}
+        removeSync(fullPath)
+    } catch (e) {
+        console.error(`Failed to remove path ${fullPath}`, e)
+    }
 }
 
-async function getSamCodeLenses(documentUri: vscode.Uri): Promise<vscode.CodeLens[]> {
+async function getDebugLocalCodeLens(documentUri: vscode.Uri, language: Language): Promise<vscode.CodeLens> {
+    return getLocalCodeLens(documentUri, language, true)
+}
+
+async function getRunLocalCodeLens(documentUri: vscode.Uri, language: Language): Promise<vscode.CodeLens> {
+    return getLocalCodeLens(documentUri, language, false)
+}
+
+async function getLocalCodeLens(documentUri: vscode.Uri, language: Language, debug: boolean): Promise<vscode.CodeLens> {
     while (true) {
         try {
             // this works without a sleep locally, but not on CodeBuild
```

Comment:
I wonder if this is still true with the changes

Reply:
I'm unsure. I'm inclined to leave this as-is at this time.